### PR TITLE
NXBT-3430: Use the Standard cookie specification

### PR DESCRIPTION
**Pull request by glefevre**

### CODE REVIEW

**Review by yjulienne-nuxeo — APPROVED** [review]
👍

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+ * (C) Copyright 2020 Nuxeo SA (http://nuxeo.com/).
```
Comment (troger) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java]: Should be the public copyright here, right?

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.http.client.config.CookieSpecs;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.nuxeo.connect.connector.http.ProxyHelper;
+
+/**
+ * Helper to define the creation of the HttpClientBuilder at only one location.
+ */
+public class HttpClientBuilderHelper {
```
Comment (troger) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java:37]: Worth adding a `@since`?

**Review by troger — COMMENTED** [review]
I am wondering if other `CloseableHttpClient httpClient = HttpClients.createDefault()` shouldn't be replaced by the new code as well?

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java (hunk 2 above)
Comment (glefevre) [reply nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java:37]: Indeed

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java (hunk 1 above)
Comment (glefevre) [reply nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java]: Done

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java (hunk 2 above)
Comment (glefevre) [reply nuxeo-connect-client/src/main/java/org/nuxeo/connect/HttpClientBuilderHelper.java:37]: Done

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/registration/RegistrationHelper.java [hunk 3]
```diff
@@ -160,7 +165,8 @@ public static String remoteRegisterInstance(String login, String password, Strin
         } catch (UnsupportedEncodingException e) {
             throw new RuntimeException(e);
         }
-        try (CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpClientBuilder httpClientBuilder = getHttpClientBuilder(null, null, url);
+        try (CloseableHttpClient httpClient = httpClientBuilder.build();
```
Comment (troger) [inline nuxeo-connect-client/src/main/java/org/nuxeo/connect/registration/RegistrationHelper.java]:
```suggestion
        try (CloseableHttpClient httpClient = getHttpClientBuilder(null, null, url).build();
```
To be consistent maybe :D

**Review by troger — APPROVED** [review]

File: nuxeo-connect-client/src/main/java/org/nuxeo/connect/registration/RegistrationHelper.java (hunk 3 above)
Comment (glefevre) [reply nuxeo-connect-client/src/main/java/org/nuxeo/connect/registration/RegistrationHelper.java]: Ah yes, I missed that

**Pull request merged**